javascript/javascript1/week3/index.js

```diff
@@ -0,0 +1,205 @@
+let totalTimeForThirdSeries =(seriesDurations.first["days"]* (seriesDurations.third["hours"]+seriesDurations.third["minutes"]/60));
+//console.log(totalTimeForFirstSeries);
+//console.log(totalTimeForSecondSeries)
+//console.log(totalTimeForThirdSeries)
+let inTotalTimeForSeries = totalTimeForFirstSeries + totalTimeForSecondSeries + totalTimeForThirdSeries;
+
+// Function to calculate percentage
+function logOutSeriesText(time,year){
+    percentage = time/year *100;
+    return percentage;
+
+}
+const firstSeries = logOutSeriesText(totalTimeForFirstSeries, myLifeSpan);
+const secondSeries = logOutSeriesText(totalTimeForSecondSeries, myLifeSpan);
+const thirdSeries = logOutSeriesText(totalTimeForThirdSeries, myLifeSpan);
+const inTotal = logOutSeriesText(inTotalTimeForSeries, myLifeSpan)
+console.log(`Game of thrones took${firstSeries}% of my life`);
+console.log(`Sopranos took${secondSeries}% of my life`);
+console.log(`The Wire took${thirdSeries}% of my life`);
+console.log(`In total that is${inTotal}% of my life`);
+
+// Step3: Note taking app!
+
+//3.1 save a note
+const notes = [];
+function saveNote(content, id){
+    let note = {
+        content: content,
+        id: id
+    };
+    return notes.push(note);
+}
+saveNote("Pickup groceries", 1);
+saveNote(" Do laundry", 2);
+console.log(notes);
+
+// 3.2 Get a note
+
+function getNote(id){
+    for(let i = 0; i<notes.length; i++){
+        if(id === isNaN() ||id ==='' ){
+            return  'please write a number in ID'
+        }
+        else if(id === (notes[i].id)){
+            return notes[i];
+        }
+        
+    }
+
+
+}
+const firstNote = getNote(1);
+console.log(firstNote);
+
+
+//3.3 log out notes
+
+function logOutNotesFormatted(){
+    for(i=0; i<notes.length; i++){
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`);
+        
+    }
+}
+logOutNotesFormatted();
+
+// Adding a activity and Usage Limit
+
+const activities = [];
+
+function addActivity(date, activity, duration){
+    let addedActivities = {date: new Date(), activity: activity, duration: duration};
+    if (typeof activity === "string" && !isNaN(duration && activity!="")){
+        return activities.push(addedActivities);
+    }
+    else{
+        return console.log("activity should be string and duration is number");
+    }
+    
+    /*activities.date = new Date();
+    activities.activity = activity;
+    activities.duration = duration;
+    activities.push({date, activity, duration});*/
+}
+
+addActivity("", "Javascript function", 30);
+addActivity("", "Fundamental of JS", 50);
+addActivity("", "Why Javascript1", 80);
+addActivity("", "Why Javascript", 20); // this is not calculate limit for only 3 activities.
+console.log(activities);
+
+
+// Show my status and Usage Limit
+
+
+
+function showStatus(){
+    let sum = 0;
+    let count = 0;
+    activities.length = 3; // usage limit so user could not get more than 3 activities.
```

Comment:
Why this limitation?

Reply:
We should never manually set the length of an array unless there's a reason to do that. What happens when you do this, is that you throw away or delete anything from index 3 on up

Reply:
I use maxLimit variable and removed activities.length